scripts/gen_type_asserts.sh

```diff
@@ -42,17 +42,44 @@ EOF
 # Create the file and initialize it with the specified content
 echo -e "$INITIAL_CONTENT" >"$ASSERT_FILE_PATH"
 
+declare -A assert_functions
+pattern="\/\/ (\w+) checks(.|\n)+?\n\}\n"
+shell_pattern="\/\/ (\w+) checks[^\/?]+?"
```

Comment:
if you don't have any variable substitutions going on, can you use simple quotes to avoid escaping those chars?

Reply:
I use online tools to create and validate regex, so it makes it easier if the escapes are there.